src/plugins/chart_expressions/expression_xy/common/utils/log_datatables.ts

```diff
@@ -45,6 +53,35 @@ export const logDatatables = (
     const logTable = prepareLogTable(layer.table, layerDimensions, true);
     handlers.inspectorAdapters.tables.logDatatable(layer.layerId, logTable);
   });
+  if (annotations) {
+    annotations.layers.forEach((layer) => {
+      const logTable = getLogAnnotationTable(annotations.datatable, layer);
+      handlers.inspectorAdapters.tables.logDatatable(layer.layerId, logTable);
+    });
+  }
+};
+
+const getLogAnnotationTable = (data: Datatable, layer: AnnotationLayerConfigResult) => {
+  const layerDimensions: Dimension[] = [
+    [['label'], strings.getLabelLabel()],
+    [['time'], strings.getTimeLabel()],
+  ];
+  const layerAnnotationsId = layer.annotations.map((annotation) => annotation.id);
```

Comment:
nit: It's on the edge of being relevant, but there might be more than a thousand rows in that table  - filtering that with an `includes` can get relatively expensive. Let's turn `layerAnnotationsId` into a `Set` to make sure it stays fast.

Reply:
Not sure if I understand it completely. Layer.annotations is not the actual datatable but the annotations config so `const layerAnnotationsId = layer.annotations.map((annotation) => annotation.id);`  gives us already unique results. How is changing an array to Set helping here?

Reply:
It's not about uniqueness, but speed - in the row filter you are doing a `find` on `layerAnnotationsId` for every row - using a set will turn a linear complexity into a constant one.